src/components/file-dropper/file-dropper.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from 'react';
+import { FileError, FileRejection, useDropzone } from 'react-dropzone';
+import styled, { css } from 'styled-components';
+import { theme } from '../layout/theme';
+import { Illustration, IllustrationSize } from '../illustration/illustration';
+import { IllustrationType } from '../illustration/illustration-set';
+import { Trans } from 'gatsby-plugin-react-i18next';
+import { TextStyles } from '../typography/typography-v2';
+import { FileListItem } from './file-list-item';
+
+interface FileDropperType {
+  file: File;
+  fileCategory: string | null;
+}
+
+interface FileDropperProps {
+  /**
+   * Called when files are uploaded and all conditions (e.g. `acceptedFileTypes` or `maxFiles`) are fulfilled or when a file changes (e.g. files gets removed or a fileCategory is added to a file). `acceptedFiles` always contains all valid files.
+   */
+  onFileChange: (acceptedFiles: FileDropperType[]) => any;
+  /**
+   * (Optional) Called when files are uploaded and a condition (e.g. wrong file type) is not fulfilled.
+   */
+  onDropRejected?: (rejectedData: FileRejection[]) => any;
+  /**
+   * (Optional) Allows custom validation of uploaded files. If the file is accepted, the `validator()` function must return null. Otherwise the function returns a FileError object. See also: [react-dropzone: custom validation](https://react-dropzone.js.org/#section-custom-validation)
+   */
+  validator?: (file: File) => FileError | FileError[] | null;
+  /**
+   * (Optional) comma-separated list of unique content type specifiers (e.g. `'image/jpeg, image/png, .pdf'`) or `undefined` to allow all files. See also: [react-dropzone: accepting specific file types](https://react-dropzone.js.org/#section-accepting-specific-file-types)
+   */
+  acceptedFileTypes?: string;
+  /**
+   * (Optional) Set a maximum number of files that can be uploaded. If this number is exceeded, `onDropRejected()` is called with all uploaded files.
+   */
+  maxFiles?: number;
+  /**
+   * (Optional) Categories that the user can add to each uploaded file (e.g. 'CV' or 'Cover Letter')
+   * */
+  fileCategories?: string[];
+  /**
+   * (Optional) Pass in a valid illustration keyword to show the according illustration.
+   */
+  illustration?: IllustrationType;
+}
```

Comment:
So you have basically checked what we are using right now from the dropzone handling in a rather uncoordinated way (https://github.com/satellytes/satellytes.com/blob/main/src/components/career-form/career-file-upload.tsx) or what was your approach?

Reply:
Yes, I checked the current file-upload component and also had some thoughts about it: The logic in the current component is only made for the use case "career form".  My plan was to make the new FileDropper component more universal so that it can be used in other places in the future if necessary. All the additional logic used in the current file-upload component will be applied in the new component through the parent.